ext/standard/formatted_print.c

```diff
@@ -247,8 +247,16 @@ php_sprintf_appenddouble(zend_string **buffer, size_t *pos,
 
 	if (zend_isinf(number)) {
 		is_negative = (number<0);
-		php_sprintf_appendstring(buffer, pos, "INF", 3, 0, padding,
-								 alignment, 3, is_negative, 0, always_sign);
+		if (is_negative) {
+			// ideally negative should have been handled inside php_sprintf_appendstring ,
+			// but the code is difficult to debug, and it's easy to break stuff inside there (I tried, I broke stuff),
+			// handling it here is much easier..
+			php_sprintf_appendstring(buffer, pos, "-INF", 4, 0, padding,
```

Comment:
Hard-coding the `min_width` argument (`4`) here, looks wrong to me (I am aware that this already had been done this way before); see https://3v4l.org/WtHcq. Why not simply pass `width` instead? Same for the `NaN` and the `-INF` cases.